cpp/src/parquet/stream_reader_test.cc

```diff
@@ -465,5 +531,151 @@ TEST_F(TestStreamReader, SkipColumns) {
+                                                 Type::INT32, ConvertedType::UINT_16));
+
+    fields.push_back(schema::PrimitiveNode::Make("int32_field", Repetition::OPTIONAL,
+                                                 Type::INT32, ConvertedType::INT_32));
+
+    fields.push_back(schema::PrimitiveNode::Make("uint64_field", Repetition::OPTIONAL,
+                                                 Type::INT64, ConvertedType::UINT_64));
+
+    fields.push_back(schema::PrimitiveNode::Make("chrono_microseconds_field",
+                                                 Repetition::OPTIONAL, Type::INT64,
+                                                 ConvertedType::TIMESTAMP_MICROS));
+
+    fields.push_back(schema::PrimitiveNode::Make("float_field", Repetition::OPTIONAL,
+                                                 Type::FLOAT, ConvertedType::NONE));
+
+    fields.push_back(schema::PrimitiveNode::Make("double_field", Repetition::OPTIONAL,
+                                                 Type::DOUBLE, ConvertedType::NONE));
+
+    return std::static_pointer_cast<schema::GroupNode>(
+        schema::GroupNode::Make("schema", Repetition::REQUIRED, fields));
+  }
+
+  void createTestFile() {
+    std::shared_ptr<arrow::io::FileOutputStream> outfile;
+
+    PARQUET_THROW_NOT_OK(arrow::io::FileOutputStream::Open(GetDataFile(), &outfile));
+
+    StreamWriter os{ParquetFileWriter::Open(outfile, GetSchema())};
+
+    TestData::init();
+
+    for (auto i = 0; i < TestData::num_rows; ++i) {
+      os << TestData::GetOptBool(i);
+      os << TestData::GetOptString(i);
+      os << TestData::GetOptChar(i);
+      os << TestData::GetOptCharArray(i);
+      os << TestData::GetOptInt8(i);
+      os << TestData::GetOptUInt16(i);
+      os << TestData::GetOptInt32(i);
+      os << TestData::GetOptUInt64(i);
+      os << TestData::GetOptChronoMicroseconds(i);
+      os << TestData::GetOptFloat(i);
+      os << TestData::GetOptDouble(i);
+      os << EndRow;
+    }
+  }
+
+  StreamReader reader_;
+};
+
+TEST_F(TestOptionalFields, ValueChecking) {
+  optional<bool> opt_bool;
+  optional<std::string> opt_string;
+  optional<std::array<char, 4>> opt_char_array;
+  optional<char> opt_char;
+  optional<int8_t> opt_int8;
+  optional<uint16_t> opt_uint16;
+  optional<int32_t> opt_int32;
+  optional<uint64_t> opt_uint64;
+  optional<std::chrono::microseconds> opt_ts_us;
+  optional<float> opt_float;
+  optional<double> opt_double;
+
+  int i;
+
+  for (i = 0; !reader_.eof(); ++i) {
+    ASSERT_EQ(i, reader_.current_row());
+
+    reader_ >> opt_bool;
+    reader_ >> opt_string;
+    reader_ >> opt_char;
+    reader_ >> opt_char_array;
+    reader_ >> opt_int8;
+    reader_ >> opt_uint16;
+    reader_ >> opt_int32;
+    reader_ >> opt_uint64;
+    reader_ >> opt_ts_us;
+    reader_ >> opt_float;
+    reader_ >> opt_double;
+    reader_ >> EndRow;
+
+    ASSERT_EQ(opt_bool, TestData::GetOptBool(i));
```

Comment:
I don't remember if i commented on the last review but EXPECT_EQ might be better here, as it allows execution to collect errors instead of exiting right away.  Given the way this is currently tested it might give messages that are too verbose as one would expect for any given column all values to fail.

Also 
`ASSET_EQ(...) << "index: " <<  i;` 

should work and given a message that include the index which would be helpful for debugging.

Reply:
Ok